ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -729,6 +753,62 @@ export default class ConfirmTransactionBase extends Component {
     );
   }
 
+  ///: BEGIN:ONLY_INCLUDE_IN(flask)
+  renderInsight() {
+    const { txData, insightSnaps } = this.props;
+    const { selectedInsightSnapId } = this.state;
+    const { txParams, chainId } = txData;
+
+    const selectedSnap = insightSnaps.find(
+      ({ id }) => id === selectedInsightSnapId,
+    );
+
+    const networkId = CHAIN_ID_TO_NETWORK_ID_MAP[chainId];
+    const caip2ChainId = `eip155:${networkId ?? stripHexPrefix(chainId)}`;
+
+    if (
+      txData.type !== TRANSACTION_TYPES.CONTRACT_INTERACTION ||
+      !insightSnaps
+    ) {
+      return null;
+    }
+
+    const dropdownOptions = insightSnaps.reduce(
+      (prev, acc) => [
+        ...prev,
+        { value: acc.id, name: acc.manifest.proposedName },
```

Comment:
also, this returns an array. It could be `insightSnaps.map(({id, manifest: { proposedName }}) => ({ value: id, name: proposedName }))`;